dcos/subcommand.py

```diff
@@ -324,6 +408,99 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _copy_and_cleanup(src, dst):
+    """Copy file or dst src to the file or directory dst and remove dst if failure
+
+    :param src: source file
+    :type src: str
+    :param dst: destination file or directory
+    :type dst: str
+    :returns: None
+    """
+
+    try:
+        if os.path.isfile(src):
+            util.sh_copy(src, dst)
+        else:
+            shutil.copytree(src, dst)
+    except Exception as e:
+        logger.exception(e)
+        shutil.rmtree(dst)
+        raise
+
+
+def _install_with_binary(
+        package_name,
+        package_dir,
+        binary_cli):
+    """
+    :param package_name: the name of the package
+    :type package_name: str
+    :param package_dir: the path to the directory in which to install the
+                          package's binary cli
+    :type package_dir: str
+    :param binary_cli: binary cli to install
+    :type binary_cli: str
+    :rtype: None
+    """
+
+    binary_url, kind = binary_cli.get("url"), binary_cli.get("kind")
+
+    try:
+        env_dir = os.path.join(package_dir,
+                               constants.DCOS_SUBCOMMAND_ENV_SUBDIR)
```

Comment:
We should just pass `env_dir` into this function and make it the same as the `pip` implementation.